hack/run-lint.sh

```diff
@@ -20,31 +20,53 @@ set -euo pipefail
 
 gopath="$(go env GOPATH)"
 
-if ! [[ -x "$gopath/bin/golangci-lint" ]]
-then
-   echo >&2 'Installing golangci-lint'
-   curl --silent --fail --location \
-       https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b "$gopath/bin" v1.17.1
+if ! [[ -x "$gopath/bin/golangci-lint" ]]; then
+  echo >&2 'Installing golangci-lint'
+  curl --silent --fail --location \
+    https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b "$gopath/bin" v1.17.1
 fi
 
 # configured by .golangci.yml
 GO111MODULE=on "$gopath/bin/golangci-lint" run
 
 install_impi() {
-   impi_dir="$(mktemp -d)"
-   trap 'rm -rf -- ${impi_dir}' EXIT
-   
-   GOPATH="${impi_dir}" \
-   GO111MODULE=off \
-   GOBIN="${gopath}/bin" \
-   go get github.com/pavius/impi/cmd/impi
+  impi_dir="$(mktemp -d)"
+  trap 'rm -rf -- ${impi_dir}' EXIT
+
+  GOPATH="${impi_dir}" \
+    GO111MODULE=off \
+    GOBIN="${gopath}/bin" \
+    go get github.com/pavius/impi/cmd/impi
 }
 
 # install impi that ensures import grouping is done consistently
-if ! [[ -x "${gopath}/bin/impi" ]]
-then
-   echo >&2 'Installing impi'
-   install_impi
+if ! [[ -x "${gopath}/bin/impi" ]]; then
+  echo >&2 'Installing impi'
+  install_impi
 fi
 
 "$gopath/bin/impi" --local sigs.k8s.io/krew --scheme stdThirdPartyLocal ./...
+
+install_shfmt() {
+  shfmt_dir="$(mktemp -d)"
+  trap 'rm -rf -- ${shfmt_dir}' EXIT
+
+  GOPATH="${shfmt_dir}" \
+    GO111MODULE=off \
+    GOBIN="${gopath}/bin" \
+    go get mvdan.cc/sh/cmd/shfmt
+}
+
+# install shfmt that ensures consistent format in shell scripts
+if ! [[ -x "${gopath}/bin/shfmt" ]]; then
+  echo >&2 'Installing shfmt'
+  install_shfmt
+fi
+
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+shfmt_help() {
+  echo "To fix run: shfmt -w -i=2 ${SCRIPTDIR}"
+}
+
+trap 'shfmt_help' ERR
+"$gopath/bin/shfmt" -d -i=2 "${SCRIPTDIR}"
```

Comment:
@ferhatelmas  let's do the following instead because (1) trap here is unnecessary (2) diff can be pretty large:

```sh
shfmt_out="$("$gopath/bin/shfmt" -l -i=2 "${SCRIPTDIR}")
if [[ -n "${shfmtout}" ]]; fi
  echo >&2 "The following shell scripts need to be formatted, run: 'shfmt -w -i=2 ${SCRIPTDIR}'
  echo >&2 "${shfmt_out}"
  exit 1
fi
```

`shfmt -l` just lists the files, and exits with code=0, so it's easier.

Reply:
LGTM, will update.

But trap was needed if diff was printed because I wanted to show run command at the end after diff (exiting).

Reply:
Updated. 

However, after second thought, since we require lint, diff in the previous implementation should have been very small incrementally.

Reply:
Valid point, although the diffs can be harder to navigate especially when a new script is added.